File: Lab07/src/Die.java
import java.util.*;
public class Die {

    private int numSides; // Number of sides the Die has
    private int currentValue; // Top value of the Die

    /**
     * Die object constructor
     * @param sides number of sides the die has
     */
    Die(int sides){

        numSides = sides;
        currentValue = (int) Math.random() * numSides;

    }

    /**
     * default Die object constructor
     * default number of sides is 6
     */
    Die(){

        numSides = 6; //default number of sides
        currentValue =  (int) Math.random() * numSides;

    }


    /**
     * rolls the Die
     * sets current value to a new value in the range o 1 to numSides
     */
    public void roll(){
        Random rand = new Random();
        int newValue = rand.nextInt(numSides);
        currentValue = newValue + 1;
    }

    /**
     * getter method for current value
     * @return Current value of this Die Object
     */
    public int getValue(){
       return currentValue;
    }
}

File: Lab04/die.py
import random


class Die:
    """
    instance of a die
    """
    def __init__(self, sides=6):
        """
        Initializes an instance of a die object

        :param sides: number of sides of the die. 6 sides by default
        """
        self.__sides = sides
        self.__top_value = random.randint(1, sides)


    def get_sides(self):
        """
        getter method for number of sides of the die

        :return: number of sides
        """
        return self.__sides

    def get_value(self):
        """
        getter method for tip value of the die

        :return: top value
        """
        return self.__top_value



    def roll(self):
        """
        rolls the die, setting the top value as a random number in the range of the die's number of sides

        :return: none
        """
        sides = self.get_sides()
        self.__top_value = random.randint(1, sides)
